aiida/tools/importexport/archive/common.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Shared resources for the archive."""
+import dataclasses
+import os
+from typing import Dict, List, Optional, Set
+
+__all__ = ('ArchiveMetadata', 'detect_archive_type')
+
+
+@dataclasses.dataclass
+class ArchiveMetadata:
+    """Class for storing metadata about this archive."""
```

Comment:
the required fields are those that are strictly necessary for recovering the archive, i.e. they are actually used during the import process.
The others are just "nice to have" and describe the parameters used to create the archive. I don't think they should default to the CLI, because this is misleading in implying that these values are contained within the archive. The CLI defaults could change between aiida versions and this also assumes that the archive was created via the CLI which is not strictly necessary.

Reply:
Makes sense!
Could you add a short version of this to the docstring?